core/server/master/src/main/java/alluxio/master/meta/conf/ConfigRecorder.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta.conf;
+
+import alluxio.wire.ConfigProperty;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class is responsible for containing server-side configurations.
+ */
+public class ConfigRecorder {
+  /** Map from ids to lockers. */
+  private ConcurrentHashMap<Long, Object> mLockers;
```

Comment:
Could we just make every method synchronized? Configuration change is a rare event, and all of these methods are fast

Reply:
Change to
```
private Object mLocker;
synchronized (mLocker)  in all methods
```

Use one object to synchronize all the methods inside a ConfigRecorder.

Reply:
You don't need a separate lock to do this, you can just use `synchronized (this) {`,  or even simpler, add the `synchronized` keyword to the method signatures. These are all the same:
```java
  private Object mLock;
  public void method() {
    synchronized (mLock) {
      doSomething();
    }
  }
```
```java
  public void method() {
    synchronized (this) {
      doSomething();
    }
  }
```
```java
  public synchronized void method() {
    doSomething();
  }
```